lib/logsink.js

```diff
@@ -0,0 +1,189 @@
+import npmlog from 'npmlog';
+import { patchLogger } from 'appium-logger';
+import winston  from 'winston';
+import fs from 'fs';
+import 'date-utils'; // TODO: make --localtimezone work without it
+
+// set up distributed logging before everything else
+patchLogger(npmlog);
+global._global_npmlog = npmlog;
+
+// npmlog is used only for emitting, we use winston for output
+npmlog.level = "silent";
+let levels = {
+  debug: 1
+, info: 2
+, warn: 3
+, error: 4
+};
+
+let colors = {
+  info: 'cyan'
+, debug: 'grey'
+, warn: 'yellow'
+, error: 'red'
+};
+
+let npmToWinstonLevels = {
+  silly: 'debug'
+, verbose: 'debug'
+, debug: 'debug'
+, info: 'info'
+, http: 'info'
+, warn: 'warn'
+, error: 'error'
+};
+
+let logger = null;
+let timeZone = null;
+
+// Capture logs emitted via npmlog and pass them through winston
+npmlog.on('log', (logObj) => {
+  let winstonLevel = npmToWinstonLevels[logObj.level] || 'info';
+  let msg = [];
+  if (logObj.prefix) msg.push(('[' + logObj.prefix + ']').magenta);
+  msg.push(logObj.message);
+  logger[winstonLevel](msg.join(' '));
+});
+
+let timestamp = () => {
+  let date = new Date();
+  if (!timeZone) {
+    date = new Date(date.valueOf() + date.getTimezoneOffset() * 60000);
+  }
+  return date.toFormat("YYYY-MM-DD HH24:MI:SS:LL");
+};
+
+// Strip the color marking within messages.
+// We need to patch the transports, because the stripColor functionality in
+// Winston is wrongly implemented at the logger level, and we want to avoid
+// having to create 2 loggers.
+function applyStripColorPatch(transport) {
```

Comment:
We don't usually put space after function name. https://github.com/appium/appium/blob/master/lib/server/helpers.js#L290